File: school-management-frontend/src/services/pedagogieService/TeacherAssignmentService.js
import api from '../api'; // ✅ MODIFICATION : Utilisation de l'instance centralisée

const TeacherAssignmentService = {
    assignTeacher: async (dto) => {
        const response = await api.post('/teacher-assignments', dto);
        return response.data;
    },
    updateAssignment: async (id, dto) => {
        const response = await api.put(`/teacher-assignments/${id}`, dto);
        return response.data;
    },
    getAssignmentsByClass: async (classroomId, yearId) => {
        const response = await api.get(`/teacher-assignments/class/${classroomId}/${yearId}`);
        return response.data;
    },
    getAssignmentsByTeacher: async (teacherId, yearId) => {
        const response = await api.get(`/teacher-assignments/teacher/${teacherId}/${yearId}`);
        return response.data;
    },
    getAssignmentById: async (id) => {
        const response = await api.get(`/teacher-assignments/${id}`);
        return response.data;
    },
    deleteAssignment: async (id) => {
        await api.delete(`/teacher-assignments/${id}`);
    },
    importPreviousYear: async (sourceYearId, targetYearId) => {
        const response = await api.post('/teacher-assignments/import-previous-year', {
            sourceYearId,
            targetYearId
        });
        return response.data;
    },
    
    // --- NOUVEAUX ENDPOINTS POUR LE TABLEAU DE BORD ---
    
    getCourseSuccessRate: async (assignmentId) => {
        const response = await api.get(`/teacher-assignments/success-rate/${assignmentId}`);
        return response.data;
    }
};

export default TeacherAssignmentService;

File: school_Management_Backend/src/main/java/com/school/management/service/academic/TeacherAssignmentService.java
package com.school.management.service.academic;

import com.school.management.dto.academic.TeacherAssignmentRequestDTO;
import com.school.management.dto.academic.TeacherAssignmentResponseDTO;

import java.util.List;

public interface TeacherAssignmentService {

    TeacherAssignmentResponseDTO assignTeacher(TeacherAssignmentRequestDTO dto);
    TeacherAssignmentResponseDTO updateAssignment(Long id, TeacherAssignmentRequestDTO dto);
    List<TeacherAssignmentResponseDTO> getAssignmentsByClass(Long classroomId, Long yearId);
    List<TeacherAssignmentResponseDTO> getAssignmentsByTeacher(Long teacherId, Long yearId);
    void deleteAssignment(Long id);

    // Nouvelle méthode pour le report d'une année à l'autre
    void importAssignmentsFromPreviousYear(Long sourceYearId, Long targetYearId);

    // NOUVELLE METHODE
    TeacherAssignmentResponseDTO getAssignmentById(Long id);

    /**
     * Calcule le taux de réussite des élèves pour un cours spécifique (affectation).
     * @param teacherAssignmentId ID de l'affectation de l'enseignant
     * @return Le pourcentage de réussite (ex: 75.0)
     */
    double getCourseSuccessRate(Long teacherAssignmentId);
}